### DISCUSSION TITLE
Are multiple Nuxt apps under one domain an anti-pattern?

### DISCUSSION THREAD

**Discussion by caseycarroll** [nuxt/nuxt#19726] [Questions]
Say an e-commerce organization has the following urls:

- shop.com/search
- shop.com/home
- shop.com/cart

...and each url is a separate project with its own sub-routes. Each project contains the backend and frontend code for said url/experience. 

Is this structure compatible with Nuxt? My understanding is a domain (shop.com) should have one instance of Nuxt. After digging around this discussion board, it seems one instance of Nuxt must exist and each project should be integrated into this Nuxt app with `extends`. 

I assume multiple Nuxt apps under one domain is wasteful and works against Nuxt's performance features. 

Are these assumptions correct?

**Comment by TheAlexLichter** [comment]
Hey 👋🏻

IMO, the assumptions are correct so far. The best way would be using different layers for different areas and then "merge" them in a single app that runs on the whole domain. Otherwise, you would lose the benefit of an SPA when moving from "one Nuxt instance" to "another instance".

`extends`/Layers is the way to go there - you can have "module"/"squad"/"domain"-specific code grouped and still get a full-blown consistent SPA in the end 👍🏻

**Comment by caseycarroll** [comment]
Gotcha! That makes sense to me.

What about "universal rendering"/SSR? If a site is made up of multiple "Nuxt instances" built via SSR, should they also leverage `extends`/Layers?

**Reply by TheAlexLichter** [reply]
You don't have to use layers but you can of course if it fits your needs. Also, with `routeRules`, you can mix rendering modes (SSR/SPA/prerendering and also support via SWR) easily ☺️

**Reply by caseycarroll** [reply]
What about shared assets?

Say every page pulled CSS from the same design system. If every instance were connected via Layers, would that shared CSS get generated at a single path that can be shared and cached across instances? i.e.
- Home App
   - shop.com/styles/design-system-buttons.css
 - Search App
   -  shop.com/styles/design-system-buttons.css

 Without Layers, I expect some assets would get "duplicated" across instances. i.e.
 - Home App
   - shop.com/**home**/styles/design-system-buttons.css
 - Search App
   -  shop.com/**search**/styles/design-system-buttons.css

**Reply by TheAlexLichter** [reply]
I think in this case, files with the same name would be overridden. But you can of course combine all the assets in the end, e.g. via CSS. 
If you keep using scoped styles and write it close to component, that might not even be necessary.

**Reply by bbhxwl** [reply]
> You don't have to use layers but you can of course if it fits your needs. Also, with `routeRules`, you can mix rendering modes (SSR/SPA/prerendering and also support via SWR) easily ☺️

Can RouteRules implement a micro front-end?
